core/src/main/scala/akka/persistence/r2dbc/query/scaladsl/R2dbcReadJournal.scala

```diff
@@ -218,6 +220,41 @@ final class R2dbcReadJournal(system: ExtendedActorSystem, config: Config, cfgPat
     }
   }
 
+  /**
+   * INTERNAL API
+   */
+  @InternalApi private[akka] def rejectPubSubTooFarAhead[Event](
+      maxAheadOfBacktracking: JDuration): Flow[EventEnvelope[Event], EventEnvelope[Event], NotUsed] = {
+    Flow[EventEnvelope[Event]]
+      .statefulMapConcat(() => {
+        // track backtracking offset
+        var latestBacktracking = Instant.EPOCH
+        env => {
+          env.offset match {
+            case t: TimestampOffset =>
+              if (env.eventOption.isEmpty) {
+                latestBacktracking = t.timestamp
+                env :: Nil
+              } else if (t.timestamp == t.readTimestamp && JDuration
```

Comment:
Only envelopes originating from pub-sub will have the same read timestamp as write? (maybe add a comment saying that)

Reply:
Yes, that is the check we have already in `R2dbcOffsetStore`. I agree that is rather vague. Would be good with something more explicit defining that its one of: backtracking, normal, pubsub, ...